### PR TITLE
Adding DateTimeOffset missing members

### LINKED ISSUE #10075: Port missing DateTime and DateTimeOffset members
These are widely used. @sokket is looking at System.Globalization.* and #10063 may be relevant to the constructors taking Calendar.

```
M:System.DateTime.#ctor(System.Int32,System.Int32,System.Int32)
M:System.DateTime.#ctor(System.Int32,System.Int32,System.Int32,System.Globalization.Calendar)
M:System.DateTime.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32)
M:System.DateTime.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Globalization.Calendar)
M:System.DateTime.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.DateTimeKind)
M:System.DateTime.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32)
M:System.DateTime.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Globalization.Calendar)
M:System.DateTime.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Globalization.Calendar,System.DateTimeKind)
M:System.DateTime.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.DateTimeKind)
M:System.DateTime.#ctor(System.Int64)
M:System.DateTime.#ctor(System.Int64,System.DateTimeKind)
M:System.DateTime.CompareTo(System.Object)
M:System.DateTime.FromOADate(System.Double)
M:System.DateTime.GetTypeCode
M:System.DateTime.System#Runtime#Serialization#ISerializable#GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
M:System.DateTime.ToLongDateString
M:System.DateTime.ToLongTimeString
M:System.DateTime.ToOADate
M:System.DateTime.ToShortDateString
M:System.DateTime.ToShortTimeString
M:System.DateTimeOffset.#ctor(System.DateTime)
M:System.DateTimeOffset.#ctor(System.DateTime,System.TimeSpan)
M:System.DateTimeOffset.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Globalization.Calendar,System.TimeSpan)
M:System.DateTimeOffset.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.TimeSpan)
M:System.DateTimeOffset.#ctor(System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.TimeSpan)
M:System.DateTimeOffset.#ctor(System.Int64,System.TimeSpan)
M:System.DateTimeOffset.System#Runtime#Serialization#IDeserializationCallback#OnDeserialization(System.Object)
M:System.DateTimeOffset.System#Runtime#Serialization#ISerializable#GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)

```

### PR DESCRIPTION
**Pull request by joperezr**
Fixes #10075

Adding the few missing members to DateTimeOffset

cc: @danmosemsft @weshaggard @stephentoub

### CODE REVIEW

**Comment by joperezr** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test please

**Comment by joperezr** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test please

**Comment by danmosemsft** [comment]
can you please include tests in the same commit?

**Comment by joperezr** [comment]
sure, I can add a test for the constructor, @stephentoub I don't think we have been adding tests for `OnDeserialization(object sender)` or `GetObjectData(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context)` members for methods that we are making serializable again, have we?

**Comment by stephentoub** [comment]
> I don't think we have been adding tests for OnDeserialization(object sender) or GetObjectData(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) members for methods that we are making serializable again, have we?

For types with custom serialization implementations, I've been adding tests that serialize/deserialize the types.  Some of these are in their own projects, and eventually I'd like each project to do its own serialization tests in addition to any we have in the BinaryFormatter test suite.  There is currently coverage of DateTimeOffset here:
https://github.com/dotnet/corefx/blob/dev/api/src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs#L73
We should also add serialization tests to the System.Runtime project for all types exposed via System.Runtime.

**Comment by joperezr** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test please

**Comment by stephentoub** [comment]
LGTM

**Comment by stephentoub** [comment]
Now that https://github.com/dotnet/corefx/pull/11316 is merged, it looks like this is causing some errors due to more members needing to be suppressed in the compat files.

**Comment by joperezr** [comment]
Test Innerloop CentOS7.1 Debug Build and Test please
Test Innerloop Ubuntu14.04 Debug Build and Test please

File: src/System.Runtime/tests/System/DateTimeOffsetTests.netstandard1.7.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using Xunit;
+
+namespace System.Tests
+{
+    public static partial class DateTimeOffsetTests
+    {
+        [Fact]
+        public static void Ctor_Calendar_TimeSpan()
+        {
+            var dateTimeOffset = new DateTimeOffset(1,1,1,0,0,0,0,new GregorianCalendar(),TimeSpan.Zero);
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/DateTimeOffsetTests.netstandard1.7.cs]: >1,1,0,0,0,0, [](start = 54, length = 12)

Nit, is there a code formatter job that runs in CI or does it only run on demand (how)

**Comment by danmosemsft** [comment]
:shipit:

File: src/System.Runtime/tests/System/DateTimeOffsetTests.netstandard1.7.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime/tests/System/DateTimeOffsetTests.netstandard1.7.cs]: On demand in one's own machine. No integrated support.

**Pull request merged**